From c85e5d3ff38abbb539b24b149d69385cb1fd29ed Mon Sep 17 00:00:00 2001
From: Oli Lan <olilan@google.com>
Date: Tue, 19 Jul 2022 10:46:00 +0000
Subject: [PATCH] Revert "Prevent non-admin users from deleting system apps."

This reverts commit 6c870e157994519094e9e50ddf93e57a26779e22.

Reason for revert: Regression, DELETE_SYSTEM_APP flag no longer works

Change-Id: Id3eb9e08a5404e88c39235d0d47337ed41bc6139
Merged-In: I4e959e296cca9bbdfc8fccc5e5e0e654ca524165
(cherry picked from commit d9089fbe06e77f5ea1773f5d69b641a81e0b5832)
Merged-In: Id3eb9e08a5404e88c39235d0d47337ed41bc6139
---
 .../com/android/server/pm/PackageManagerService.java   | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/services/core/java/com/android/server/pm/PackageManagerService.java b/services/core/java/com/android/server/pm/PackageManagerService.java
index 59e74f36c744..f9c40a3eb68c 100644
--- a/services/core/java/com/android/server/pm/PackageManagerService.java
+++ b/services/core/java/com/android/server/pm/PackageManagerService.java
@@ -21532,16 +21532,6 @@ public class PackageManagerService extends IPackageManager.Stub
                 return PackageManager.DELETE_FAILED_INTERNAL_ERROR;
             }
 
-            if (isSystemApp(uninstalledPs)) {
-                UserInfo userInfo = mUserManager.getUserInfo(userId);
-                if (userInfo == null || !userInfo.isAdmin()) {
-                    Slog.w(TAG, "Not removing package " + packageName
-                            + " as only admin user may downgrade system apps");
-                    EventLog.writeEvent(0x534e4554, "170646036", -1, packageName);
-                    return PackageManager.DELETE_FAILED_USER_RESTRICTED;
-                }
-            }
-
             disabledSystemPs = mSettings.getDisabledSystemPkgLPr(packageName);
             // Static shared libs can be declared by any package, so let us not
             // allow removing a package if it provides a lib others depend on.
-- 
2.38.1.273.g43a17bfeac-goog

